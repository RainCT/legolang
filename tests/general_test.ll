float foo = 1.1
foo = 1.2 

bool meh;

meh = 5 || 6 and foo + 3 * 6

bool grr;
grr =	!(5 > -5 and 4 != 32)
grr = not (5 > -5 and 4 != 32)

print "No position"

if foo:
	print "ifs rock"
	print "for real"
	
def funcioneta (int x=3, string s="hello") returns int:
	print 5+3
	return 1.32
	
	
		  // hola
   /* dsf
   
	*/ 
	
	def meh (int x, int y, int z=5):
		print 3
  
def noparams ():
	print "noparams";
  
int x = 1
int y = 2
int z = 3
funcioneta(x, "hi")
int A = funcioneta(foo, "testing float")
noparams()

bool a = (4 >6)

int tst = 10
while tst :
	print "Hello"
	print tst
	tst = tst - 1

def intint (int x=3) returns float :
	print "I'm an int function"
	int sieg = 10;
	while sieg > 0 :
		print sieg
		sieg = sieg - 1
		return 4+x
	print intint()
	return 1

def funception (int x) returns int :
	def firstfunc (float j) returns int :
		j = 3;
		def innerfunc (float k) returns bool :
			print "inner function"
			return true
 	 
		j = j*3;
		return j
	return firstfunc(5.0) + 32

funception(5);
if true :
	print "hello if"
else :
	print "hello els"

print "moving"
