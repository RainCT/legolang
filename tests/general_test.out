No position
ifs rock
for real
8
8
noparams
Hello
10
Hello
9
Hello
8
Hello
7
Hello
6
Hello
5
Hello
4
Hello
3
Hello
2
Hello
1
hello if
moving
